src/Configuration.php

```diff
@@ -15,7 +15,7 @@
  * Serves like a small DI container to simplify the creation and usage
  * of the objects.
  */
-final class Configuration
+final readonly class Configuration
```

Comment:
We need to open a discussion about this component... its design is a bit too convoluted for me.
I'll post something soon...

Reply:
I'm curious